internal/graph/check.go

```diff
@@ -459,8 +474,28 @@ func (c *LocalChecker) checkDirect(parentctx context.Context, req *ResolveCheckR
 			return union(ctx, c.concurrencyLimit, handlers...)
 		}
 
-		return union(ctx, c.concurrencyLimit, fn1, fn2)
+		outerHandlers = append(outerHandlers, fn2)
+
+		return union(ctx, c.concurrencyLimit, outerHandlers...)
+	}
+}
+
+// allowedTypesForUser could be "team" or "team:*" or "group#member"
+// shouldCheckDirectTuple returns true if allowedTypes contains "team" and the user is like type "team:fga"
+// or if allowedTypes contains "group#member" and the user is like type "group:fga#member"
+func (c *LocalChecker) shouldCheckDirectTuple(typesys *typesystem.TypeSystem, allowedTypes []*openfgapb.RelationReference, user string) bool {
```

Comment:
There's already a function for this in the `typesystem` package. We shouldn't add this here.

You should be able to use `typesys.IsDirectlyRelated(...)`